### DISCUSSION TITLE
What are advantages over other libraries like Handlebars, Razor?

### DISCUSSION THREAD

**Discussion by rklec** [axuno/SmartFormat#440] [Q&A]
I am currently considering which library to use for templating things in .NET.

* I also found https://github.com/Handlebars-Net/Handlebars.Net - which seems to precompile templates and use it in a good way.
* The [Razor engine](https://stackoverflow.com/a/5101333/15488899) is also suggested for this use case?

**Comment by axunonb** [comment]
Maybe the question is a bit too general to answer correctly for your use case.
If it comes to speed, small memory footprint, easy implementation, number of dependencies and a quick learning curve, `Smart.Format` looks like a good choice. 
When templating also means to handle complex conditions or logic, that requires a programming language like C# for Razor, other alternatives might be better. Also, Razor is a good way to go when targeting to HTML/CSS templates.
Hope this helps.
